### DISCUSSION TITLE
Anyone used AuthorizationHeaderValueGetter with ITokenAcquisition service?

### DISCUSSION THREAD

**Discussion by SWarnberg** [reactiveui/refit#1638] [Q&A] [answered]
I was trying to use the **AuthorizationHeaderValueGetter** setting to set the token for the protected endpoints because it seems to be a good pattern. Being in ASP .NET Core, the **ITokenAcquisition** DI service would be the preferred way to get the token if you have services protected by Identity Platform.
However, wherever the function for AuthorizationHeaderValueGetter is put, the DI services are not available, and the following error is returned.
```Cannot resolve scoped service 'Microsoft.Identity.Web.ITokenAcquisition' from root provider.```

The above scenario is easily fixed with a message handler instead, but I was hoping for this functionality to be more slick.

**Comment by DanielTuran** [comment]
Just to clarify, ITokenAcquisition is registered as Scoped service, because if you want acquire access token for user, you need a scope (Scope is bound to http request, which has a user).

It seems like a bug, that Refit uses root service provider even when scope is available (e.g. injecting refit client to scoped service)

**Comment by frogcrush** [comment, marked as answer]
I have (and wrote a blog post - [link](https://tylorstech.com/2026/01/26/authenticating-with-itokenprovider-msal-and-refit-in-asp-net-core/)) but essentially what I do is create a factory that holds a method that can access the HttpContext, and from there you can pull a scoped service that allows you to access ITokenAcquisition.
